src/assets/scripts/client/aircraft/AircraftInstanceModel.js

```diff
@@ -2272,49 +1259,146 @@ export default class Aircraft {
             return;
         }
 
+        this.updateAircraftTurnPhysics();
+
+        this.updateAltitudePhysics();
+
+        if (this.isOnGround()) {
+            this.trend = 0;
+        }
+
+        // SPEED
+        this.updateSpeedPhysics();
+
+        if (!this.position) {
+            return;
+        }
+
+        // Trailling
+        if (this.position_history.length === 0) {
+            this.position_history.push([
```

Comment:
could you add this TODO comment here: `TODO: abstract to AircraftPositionHistory class`